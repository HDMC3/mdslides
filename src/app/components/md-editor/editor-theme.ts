import { EditorView } from '@codemirror/view';

const tooltipBackground = '#353a42';

const textBasicColor = 'var(--text-basic-color)';
const textHintColor = 'var(--text-hint-color)';
const backgroundBasicColor1 = 'var(--background-basic-color-1)';
const backgroundBasicColor2 = 'var(--background-basic-color-2)';
const backgroundBasicColor4 = 'var(--background-basic-color-4)';
const colorBasic800 = 'var(--color-basic-800)';
const colorBasicTransparent600 = 'var(--color-basic-transparent-600)';
const colorControlDefault = 'var(--color-control-default)';
const colorControlActive = 'var(--color-control-active)';
const colorControlHover = 'var(--color-control-hover)';

export const oneDarkTheme = EditorView.theme({
    '&': {
        color: textBasicColor,
        backgroundColor: backgroundBasicColor1
    },

    '.cm-content': {
        caretColor: textHintColor
    },

    '.cm-cursor, .cm-dropCursor': { borderLeftColor: textHintColor },
    '&.cm-focused .cm-selectionBackground, .cm-selectionBackground, .cm-content ::selection': { backgroundColor: colorBasicTransparent600 },

    '.cm-panels': { backgroundColor: backgroundBasicColor4, color: textBasicColor },
    '.cm-panels.cm-panels-top': { borderBottom: '2px solid black' },
    '.cm-panels.cm-panels-bottom': { borderTop: '2px solid black' },

    '.cm-button': { border: 'none', backgroundImage: 'none', background: colorControlDefault, color: colorBasic800 },
    '.cm-button:active': { border: 'none', backgroundImage: 'none', background: colorControlActive },
    '.cm-button:hover': { border: 'none', backgroundImage: 'none', background: colorControlHover },
    '.cm-textfield': { color: colorBasic800 },

    '.cm-panel button[name=close]': { color: textBasicColor },

    '.cm-searchMatch': {
        backgroundColor: '#72a1ff59',
        outline: '1px solid #457dff'
    },
    '.cm-searchMatch.cm-searchMatch-selected': {
        backgroundColor: '#6199ff2f'
    },

    '.cm-activeLine': { backgroundColor: '#6699ff0b' },
    '.cm-selectionMatch': { backgroundColor: '#aafe661a' },

    '&.cm-focused .cm-matchingBracket, &.cm-focused .cm-nonmatchingBracket': {
        backgroundColor: '#bad0f847',
        outline: '1px solid #515a6b'
    },

    '.cm-gutters': {
        backgroundColor: backgroundBasicColor2,
        color: textHintColor,
        border: 'none'
    },

    '.cm-activeLineGutter': {
        backgroundColor: backgroundBasicColor4
    },

    '.cm-foldPlaceholder': {
        backgroundColor: 'transparent',
        border: 'none',
        color: '#ddd'
    },

    '.cm-tooltip': {
        border: 'none',
        backgroundColor: backgroundBasicColor4
    },
    '.cm-tooltip .cm-tooltip-arrow:before': {
        borderTopColor: 'transparent',
        borderBottomColor: 'transparent'
    },
    '.cm-tooltip .cm-tooltip-arrow:after': {
        borderTopColor: tooltipBackground,
        borderBottomColor: tooltipBackground
    },
    '.cm-tooltip-autocomplete': {
        '& > ul > li[aria-selected]': {
            backgroundColor: colorBasicTransparent600,
            color: textBasicColor
        }
    },
    '.ͼb': { color: 'var(--text-primary-focus-color)' },
    '.ͼc': {
        color: textHintColor
    },
    '.ͼd': { color: 'var(--text-warning-focus-color)' },
    '.ͼe': { color: 'var(--text-danger-focus-color)' },
    '.ͼf': { color: 'var(--text-danger-hover-color)' },
    '.ͼg': { color: '#940' },
    '.ͼh': { color: '#708' },
    '.ͼi': { color: 'var(--text-warning-color)' },
    '.ͼj': { color: 'var(--text-success-color)' },
    '.ͼk': { color: 'var(--text-success-focus-color)' },
    '.ͼl': { color: 'var(--text-info-color)' },
    '.ͼm': { color: 'var(--text-primary-color)' },
    '.ͼn': { color: 'var(--text-danger-color)' }
}, { dark: false });
