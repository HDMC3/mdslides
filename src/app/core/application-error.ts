export class ApplicationError extends Error {
    constructor(message: string, public title?: string) {
        super(message);
    }
}
