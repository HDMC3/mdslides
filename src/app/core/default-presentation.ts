import { v4 } from 'uuid';
import { Presentation } from '../data/interfaces/presentation';

export const getDefaultPresentation = (id: string): Presentation => {
    return {
        id,
        title: 'Titulo de presentacion',
        creation_date: new Date(),
        modification_date: new Date(),
        description: '',
        slides: [
            {
                id: v4(),
                name: 'Presentacion inicial',
                code: [
                    '---',
                    'style: |',
                    'section {',
                    '   display: flex;',
                    '   flex-direction: column;',
                    '   align-items: center;',
                    '   justify-content: center;',
                    '}',
                    '---',
                    '',
                    '# Inicio de la presentacion',
                    '# ðŸ˜„âœŒ'
                ]
            }
        ]
    };
};
