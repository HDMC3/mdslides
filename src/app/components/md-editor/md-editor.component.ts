import { Component, ElementRef, OnInit, Output, EventEmitter } from '@angular/core';
import { markdown } from '@codemirror/lang-markdown';
import { basicSetup } from 'codemirror';
import { EditorState } from '@codemirror/state';
import { languages } from '@codemirror/language-data';
import { EditorView } from '@codemirror/view';
import { oneDarkTheme } from './editor-theme';

@Component({
    selector: 'app-md-editor',
    templateUrl: './md-editor.component.html',
    styleUrls: ['./md-editor.component.css']
})
export class MdEditorComponent implements OnInit {
    @Output() valueChange: EventEmitter<string[]>;
    editorInitialized: boolean;

    constructor(private elementRef: ElementRef) {
        this.editorInitialized = false;
        this.valueChange = new EventEmitter();
    }

    ngOnInit(): void {
        const state = EditorState.create({
            extensions: [
                basicSetup,
                markdown({ codeLanguages: languages }),
                EditorView.domEventHandlers({
                    keyup: (e, v) => {
                        const editorValue = v.state.doc.toJSON();
                        this.valueChange.emit(editorValue);
                    }
                }),
                EditorView.lineWrapping,
                oneDarkTheme
            ]
        });

        const view = new EditorView({
            state,
            parent: this.elementRef.nativeElement
        });

        view.dom.style.height = '100%';
    }

    initEditor() {
    }

}
