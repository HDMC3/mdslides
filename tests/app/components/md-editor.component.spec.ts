import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MdEditorService } from '../../../src/app/core/services/md-editor.service';

import { MdEditorComponent } from '../../../src/app/components/md-editor/md-editor.component';

describe('MdEditorComponent', () => {
    let component: MdEditorComponent;
    let fixture: ComponentFixture<MdEditorComponent>;
    let compiled: HTMLElement;

    beforeEach(async() => {
        await TestBed.configureTestingModule({
            declarations: [MdEditorComponent]
        })
            .compileComponents();
        TestBed.inject(MdEditorService);
    });

    beforeEach(() => {
        fixture = TestBed.createComponent(MdEditorComponent);
        component = fixture.componentInstance;
        fixture.detectChanges();
        compiled = fixture.nativeElement;
    });

    it('should create', () => {
        expect(component).toBeTruthy();
    });

    it('shoudl match snapshot', () => {
        expect(compiled).toMatchSnapshot();
    });
});
