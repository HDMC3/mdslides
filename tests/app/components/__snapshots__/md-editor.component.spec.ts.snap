// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MdEditorComponent shoudl match snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="cm-editor ͼ1 ͼ2 ͼ4 ͼo"
    style="height: 100%;"
  >
    <div
      aria-live="polite"
      style="position: absolute; top: -10000px;"
    />
    <div
      class="cm-scroller"
      tabindex="-1"
    >
      <div
        aria-hidden="true"
        class="cm-gutters"
        style="min-height: 28px; position: sticky;"
      >
        <div
          class="cm-gutter cm-lineNumbers"
        >
          <div
            class="cm-gutterElement"
            style="height: 0px; visibility: hidden; pointer-events: none;"
          >
            9
          </div>
          <div
            class="cm-gutterElement cm-activeLineGutter"
            style="height: 14px;"
          >
            1
          </div>
          <div
            class="cm-gutterElement"
            style="height: 14px;"
          >
            2
          </div>
        </div>
        <div
          class="cm-gutter cm-foldGutter"
        >
          <div
            class="cm-gutterElement"
            style="height: 0px; visibility: hidden; pointer-events: none;"
          >
            <span
              title="Unfold line"
            >
              ›
            </span>
          </div>
          <div
            class="cm-gutterElement cm-activeLineGutter"
            style="height: 14px;"
          />
        </div>
      </div>
      <div
        aria-autocomplete="list"
        aria-multiline="true"
        autocapitalize="off"
        autocorrect="off"
        class="cm-content cm-lineWrapping"
        contenteditable="true"
        role="textbox"
        spellcheck="false"
        style="tab-size: 4;"
        translate="no"
      >
        <div
          class="cm-activeLine cm-line"
        >
          <br />
        </div>
        <div
          class="cm-line"
        >
          <br />
        </div>
      </div>
      <div
        aria-hidden="true"
        class="cm-selectionLayer"
      />
      <div
        aria-hidden="true"
        class="cm-cursorLayer"
        style="animation-duration: 1200ms;"
      />
    </div>
  </div>
</div>
`;
